app/src/main/java/net/programmierecke/radiodroid2/StationSaveManager.java

```diff
@@ -92,6 +93,23 @@ public void move(int fromPos, int toPos) {
         Collections.rotate(listStations.subList(Math.min(fromPos, toPos), Math.max(fromPos, toPos) + 1), Integer.signum(fromPos - toPos));
     }
 
+    public @Nullable DataRadioStation getBestNameMatch(String query) {
+        DataRadioStation bestStation = null;
+        query = query.toUpperCase();
+        double smallesDistance = Double.MAX_VALUE;
+
+        for (DataRadioStation station : listStations) {
+            Cosine distMeasure = new Cosine();
```

Comment:
Probably it shouldn't be created inside the loop.

Reply:
Unfortunately at least the measures it tested this with required to be renewed every time, but I'll check if cosine (which seems to work quite well for keyboard input)  is affected as well.

Reply:
Ah, I love such library design.

Reply:
If it true it worth adding a comment.

Reply:
OK, for Cosine it works inside the loop. I've added a warning comment for other measures.